File: packages/expo-image-manipulator/android/src/main/java/expo/modules/imagemanipulator/FileUtils.java
package expo.modules.imagemanipulator;

import java.io.File;
import java.io.IOException;
import java.util.UUID;

class FileUtils {
  public static File ensureDirExists(File dir) throws IOException {
    if (!(dir.isDirectory() || dir.mkdirs())) {
      throw new IOException("Couldn't create directory '" + dir + "'");
    }
    return dir;
  }

  public static String generateOutputPath(File internalDirectory, String dirName, String extension) throws IOException {
    File directory = new File(internalDirectory + File.separator + dirName);
    FileUtils.ensureDirExists(directory);
    String filename = UUID.randomUUID().toString();
    return directory + File.separator + filename + extension;
  }
}

File: packages/expo-image-manipulator/android/src/main/java/expo/modules/imagemanipulator/FileUtils.kt
package expo.modules.imagemanipulator

import java.io.File
import java.io.IOException
import java.util.*

internal object FileUtils {
  @Throws(IOException::class)
  fun generateRandomOutputPath(cacheDirectory: File, imageFormat: ImageFormat): String {
    val directory = File(cacheDirectory, "ImageManipulator")
    ensureDirExists(directory)
    return "${directory}${File.separator}${UUID.randomUUID()}${imageFormat.fileExtension}"
  }

  @Throws(IOException::class)
  private fun ensureDirExists(dir: File): File {
    if (!(dir.isDirectory || dir.mkdirs())) {
      throw ImageWriteFailedException(dir.path)
    }
    return dir
  }
}
